app/gui2/src/util/ast/__tests__/callTree.test.ts

```diff
@@ -107,6 +121,143 @@ test.each`
   },
 )
 
+interface TestCase {
+  description: string
+  code: string
+  /** Index of sub-application for which the MethodCallInfo is available. */
+  subapplicationIndex: number
+  /** Not applied arguments in available MethodCallInfo. */
+  notAppliedArguments: number[]
+  /** Not applied arguments expected for the whole expression. */
+  expectedNotAppliedArguments: number[]
+}
+
+test.each([
+  {
+    description: 'Base case',
+    code: 'Aggregate_Column.Sum',
+    subapplicationIndex: 0,
+    notAppliedArguments: [0, 1],
+    expectedNotAppliedArguments: [0, 1],
+  },
+  {
+    description: '1 arg, info on most inner subapplication.',
+    code: 'Aggregate_Column.Sum x',
+    subapplicationIndex: 1,
+    notAppliedArguments: [0, 1],
+    expectedNotAppliedArguments: [1],
+  },
+  {
+    description: '2 args, info on most inner subapplication.',
+    code: 'Aggregate_Column.Sum x y',
+    subapplicationIndex: 2,
+    notAppliedArguments: [0, 1],
+    expectedNotAppliedArguments: [],
+  },
+  {
+    description: '2 args, info on inner subapplication.',
+    code: 'Aggregate_Column.Sum x y',
+    subapplicationIndex: 1,
+    notAppliedArguments: [1],
+    expectedNotAppliedArguments: [],
+  },
+  {
+    description: '1 arg, notAppliedArguments unsorted.',
+    code: 'Aggregate_Column.Sum x',
+    subapplicationIndex: 1,
+    notAppliedArguments: [1, 0],
+    expectedNotAppliedArguments: [1],
+  },
+  {
+    description: '1 named arg.',
+    code: 'Aggregate_Column.Sum as=x',
+    subapplicationIndex: 1,
+    notAppliedArguments: [0, 1],
+    expectedNotAppliedArguments: [0],
+  },
+  {
+    description: '2 named args.',
+    code: 'Aggregate_Column.Sum as=x column=y',
+    subapplicationIndex: 2,
+    notAppliedArguments: [0, 1],
+    expectedNotAppliedArguments: [],
+  },
+  {
+    description: '1 wrongly named arg.',
+    code: 'Aggregate_Column.Sum bla=x',
+    subapplicationIndex: 1,
+    notAppliedArguments: [0, 1],
+    expectedNotAppliedArguments: [0, 1],
+  },
+  {
+    description: '1 named & 1 unnamed args.',
+    code: 'Aggregate_Column.Sum as=x y',
+    subapplicationIndex: 2,
+    notAppliedArguments: [0, 1],
+    expectedNotAppliedArguments: [],
+  },
+] as TestCase[])(
```

Comment:
Make sure to also add a test case with `notAppliedArguments` being "incorrectly" empty, which afaik is the actual response we get from engine for constructors.

Reply:
Done